.s2i/bin/run

```diff
@@ -15,18 +15,27 @@ USER_ID=$(id -u)
 GROUP_ID=$(id -g)
 sed "s@${USER_NAME}:x:\${USER_ID}:\${GROUP_ID}@${USER_NAME}:x:${USER_ID}:${GROUP_ID}@g" ${APP_ROOT}/etc/passwd.template > /etc/passwd
 
-INVENTORY="${APP_HOME}/tmp_inventory"
-INVENTORY_ARG="-i ${INVENTORY}"
+INVENTORY="$(mktemp)"
 if [[ -v INVENTORY_FILE ]]; then
-  cp ${INVENTORY_FILE} ${INVENTORY}
+  # If the pointed inventory has execute/search perms we can assume it
+  # contains a dynamic inventory and we use it directly.
+  # Otherwise we make a copy so that ALLOW_ANSIBLE_CONNECTION_LOCAL below
+  # does not attempt to modify the original
+  if [[ -x ${INVENTORY_FILE} ]]; then
```

Comment:
I agree that it would be much better. However, it's a bit harder to make the check when the inventory is being downloaded: are we downloading a static inventory or a dynamic script? We'd have to look at the content to tell...

Maybe we can create an issue to follow up on this in the future?